Does not compile on OSX

```
brew install ctags lemon
```

```
samm-mb ~/git/miller/c % make
ctags -R .
make -C dsls put_dsl_parse.o
lemon put_dsl_parse.y
mv put_dsl_parse.c put_dsl_parse.c.tmp
sed \
			-e 's/ParseTrace/put_dsl_ParseTrace/g' \
			-e 's/ParseTokenName/put_dsl_ParseTokenName/g' \
			-e 's/lemon_parser_alloc/put_dsl_lemon_parser_alloc/g' \
			-e 's/lemon_parser_free/put_dsl_lemon_parser_free/g' \
			-e 's/lemon_parser_parse_token/put_dsl_lemon_parser_parse_token/g' \
			-e 's/yy_destructor/put_dsl_yy_destructor/g' \
		put_dsl_parse.c.tmp > put_dsl_parse.c
rm -f put_dsl_parse.c.tmp
gcc -I.. -O3 -c -std=gnu99 put_dsl_parse.c
put_dsl_parse.c:529:4: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
                 ParseARG_FETCH;
                 ^
put_dsl_parse.c:85:40: note: expanded from macro 'ParseARG_FETCH'
#define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                       ^
put_dsl_parse.c:521:18: note: 'pparser' declared here
        lemon_parser_t *pparser,          /* The parser to be shifted */
                        ^
put_dsl_parse.c:539:4: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
                 ParseARG_STORE; /* Suppress warning about unused %extra_argument var */
                 ^
put_dsl_parse.c:86:24: note: expanded from macro 'ParseARG_STORE'
#define ParseARG_STORE yypParser->pasts = pasts
                       ^
put_dsl_parse.c:521:18: note: 'pparser' declared here
        lemon_parser_t *pparser,          /* The parser to be shifted */
                        ^
put_dsl_parse.c:618:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_FETCH;
        ^
put_dsl_parse.c:85:40: note: expanded from macro 'ParseARG_FETCH'
#define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                       ^
put_dsl_parse.c:610:18: note: 'pparser' declared here
        lemon_parser_t *pparser,         /* The parser */
                        ^
put_dsl_parse.c:650:89: warning: implicit declaration of function 'yytestcase' is invalid in C99 [-Wimplicit-function-declaration]
      case 6: /* put_dsl_or_term ::= put_dsl_or_term FILTER_DSL_AND put_dsl_and_term */ yytestcase(yyruleno==6);
                                                                                        ^
put_dsl_parse.c:771:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_FETCH;
        ^
put_dsl_parse.c:85:40: note: expanded from macro 'ParseARG_FETCH'
#define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                       ^
put_dsl_parse.c:770:45: note: 'pparser' declared here
static void yy_parse_failed(lemon_parser_t *pparser) {
                                            ^
put_dsl_parse.c:781:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_STORE; /* Suppress warning about unused %extra_argument variable */
        ^
put_dsl_parse.c:86:24: note: expanded from macro 'ParseARG_STORE'
#define ParseARG_STORE yypParser->pasts = pasts
                       ^
put_dsl_parse.c:770:45: note: 'pparser' declared here
static void yy_parse_failed(lemon_parser_t *pparser) {
                                            ^
put_dsl_parse.c:792:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_FETCH;
        ^
put_dsl_parse.c:85:40: note: expanded from macro 'ParseARG_FETCH'
#define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                       ^
put_dsl_parse.c:788:18: note: 'pparser' declared here
        lemon_parser_t *pparser,           /* The parser */
                        ^
put_dsl_parse.c:798:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_STORE; /* Suppress warning about unused %extra_argument variable */
        ^
put_dsl_parse.c:86:24: note: expanded from macro 'ParseARG_STORE'
#define ParseARG_STORE yypParser->pasts = pasts
                       ^
put_dsl_parse.c:788:18: note: 'pparser' declared here
        lemon_parser_t *pparser,           /* The parser */
                        ^
put_dsl_parse.c:805:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_FETCH;
        ^
put_dsl_parse.c:85:40: note: expanded from macro 'ParseARG_FETCH'
#define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                       ^
put_dsl_parse.c:804:39: note: 'pparser' declared here
static void yy_accept(lemon_parser_t *pparser) {
                                      ^
put_dsl_parse.c:822:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_STORE; /* Suppress warning about unused %extra_argument variable */
        ^
put_dsl_parse.c:86:24: note: expanded from macro 'ParseARG_STORE'
#define ParseARG_STORE yypParser->pasts = pasts
                       ^
put_dsl_parse.c:804:39: note: 'pparser' declared here
static void yy_accept(lemon_parser_t *pparser) {
                                      ^
put_dsl_parse.c:870:2: error: use of undeclared identifier 'yypParser'; did you mean 'pparser'?
        ParseARG_STORE;
        ^
put_dsl_parse.c:86:24: note: expanded from macro 'ParseARG_STORE'
#define ParseARG_STORE yypParser->pasts = pasts
                       ^
put_dsl_parse.c:855:18: note: 'pparser' declared here
        lemon_parser_t *pparser;  /* The parser */
                        ^
1 warning and 10 errors generated.
make[1]: *** [put_dsl_parse.o] Error 1
make: *** [dsls] Error 2
```

I'm getting similar errors on Archlinux. On a clean `make`, I get:

```
[andrew@Liger c] make
ctags -R .
make -C dsls put_dsl_parse.o
make[1]: Entering directory '/home/andrew/clones/miller/c/dsls'
gcc -o lemon lemon.c
lemon.c:2277:8: warning: return type defaults to ‘int’ [-Wimplicit-int]
 static preprocess_input(char *z){
        ^
lemon.c: In function ‘tplt_open’:
lemon.c:2863:6: warning: implicit declaration of function ‘access’ [-Wimplicit-function-declaration]
  if (access(buf,004)==0) {
      ^
lemon.c:2758:13: note: previous declaration of ‘access’ was here
  extern int access();
             ^
lemon put_dsl_parse.y
make[1]: lemon: Command not found
Makefile:16: recipe for target 'put_dsl_parse.c' failed
make[1]: *** [put_dsl_parse.c] Error 127
make[1]: Leaving directory '/home/andrew/clones/miller/c/dsls'
Makefile:70: recipe for target 'dsls' failed
make: *** [dsls] Error 2
```

OK, so maybe I do need `lemon` after all... Once I install `lemon`, I get a similar output to the OP:

```
[andrew@Liger c] make
ctags -R .
make -C dsls put_dsl_parse.o
make[1]: Entering directory '/home/andrew/clones/miller/c/dsls'
gcc -o lemon lemon.c
lemon.c:2277:8: warning: return type defaults to ‘int’ [-Wimplicit-int]
 static preprocess_input(char *z){
        ^
lemon.c: In function ‘tplt_open’:
lemon.c:2863:6: warning: implicit declaration of function ‘access’ [-Wimplicit-function-declaration]
  if (access(buf,004)==0) {
      ^
lemon.c:2758:13: note: previous declaration of ‘access’ was here
  extern int access();
             ^
lemon put_dsl_parse.y
mv put_dsl_parse.c put_dsl_parse.c.tmp
sed \
                -e 's/ParseTrace/put_dsl_ParseTrace/g' \
                -e 's/ParseTokenName/put_dsl_ParseTokenName/g' \
                -e 's/lemon_parser_alloc/put_dsl_lemon_parser_alloc/g' \
                -e 's/lemon_parser_free/put_dsl_lemon_parser_free/g' \
                -e 's/lemon_parser_parse_token/put_dsl_lemon_parser_parse_token/g' \
                -e 's/yy_destructor/put_dsl_yy_destructor/g' \
        put_dsl_parse.c.tmp > put_dsl_parse.c
rm -f put_dsl_parse.c.tmp
gcc -I.. -O3 -c -std=gnu99 put_dsl_parse.c
put_dsl_parse.c: In function ‘yy_shift’:
put_dsl_parse.c:85:40: error: ‘yypParser’ undeclared (first use in this function)
 #define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                        ^
put_dsl_parse.c:532:4: note: in expansion of macro ‘ParseARG_FETCH’
    ParseARG_FETCH;
    ^
put_dsl_parse.c:85:40: note: each undeclared identifier is reported only once for each function it appears in
 #define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                        ^
put_dsl_parse.c:532:4: note: in expansion of macro ‘ParseARG_FETCH’
    ParseARG_FETCH;
    ^
put_dsl_parse.c: In function ‘yy_reduce’:
put_dsl_parse.c:85:40: error: ‘yypParser’ undeclared (first use in this function)
 #define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                        ^
put_dsl_parse.c:621:2: note: in expansion of macro ‘ParseARG_FETCH’
  ParseARG_FETCH;
  ^
put_dsl_parse.c:653:89: warning: implicit declaration of function ‘yytestcase’ [-Wimplicit-function-declaration]
       case 6: /* put_dsl_or_term ::= put_dsl_or_term FILTER_DSL_AND put_dsl_and_term */ yytestcase(yyruleno==6);
                                                                                         ^
put_dsl_parse.c: In function ‘yy_parse_failed’:
put_dsl_parse.c:85:40: error: ‘yypParser’ undeclared (first use in this function)
 #define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                        ^
put_dsl_parse.c:774:2: note: in expansion of macro ‘ParseARG_FETCH’
  ParseARG_FETCH;
  ^
put_dsl_parse.c: In function ‘yy_syntax_error’:
put_dsl_parse.c:85:40: error: ‘yypParser’ undeclared (first use in this function)
 #define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                        ^
put_dsl_parse.c:795:2: note: in expansion of macro ‘ParseARG_FETCH’
  ParseARG_FETCH;
  ^
put_dsl_parse.c: In function ‘yy_accept’:
put_dsl_parse.c:85:40: error: ‘yypParser’ undeclared (first use in this function)
 #define ParseARG_FETCH sllv_t* pasts = yypParser->pasts
                                        ^
put_dsl_parse.c:808:2: note: in expansion of macro ‘ParseARG_FETCH’
  ParseARG_FETCH;
  ^
put_dsl_parse.c: In function ‘put_dsl_lemon_parser_parse_token’:
put_dsl_parse.c:86:24: error: ‘yypParser’ undeclared (first use in this function)
 #define ParseARG_STORE yypParser->pasts = pasts
                        ^
put_dsl_parse.c:873:2: note: in expansion of macro ‘ParseARG_STORE’
  ParseARG_STORE;
  ^
Makefile:14: recipe for target 'put_dsl_parse.o' failed
make[1]: *** [put_dsl_parse.o] Error 1
make[1]: Leaving directory '/home/andrew/clones/miller/c/dsls'
Makefile:70: recipe for target 'dsls' failed
make: *** [dsls] Error 2
```

I'm getting this on Yosemite 10.10.5

CPWKRKNMDTY3:~/src/miller-master/c/dsls$ make
gcc -I.. -O3 -std=gnu99 -D __PUT_DSL_MAIN__ ../containers/sllv.c ../containers/slls.c ../lib/mlrutil.c ../containers/lrec.c ../containers/header_keeper.c put_dsl_parse.c put_dsl_lexer.c ../containers/mlr_dsl_ast.c put_dsl_wrapper.c -o pdm -lm
put_dsl_wrapper.c:24:16: warning: implicit declaration of function 'put_dsl_lemon_parser_parse_token' is invalid in C99 [-Wimplicit-function-declaration]
                parse_code = put_dsl_lemon_parser_parse_token(pvparser, lex_code, plexed_node, pasts);
                             ^
put_dsl_wrapper.c:61:2: warning: implicit declaration of function 'put_dsl_lemon_parser_free' is invalid in C99 [-Wimplicit-function-declaration]
        put_dsl_lemon_parser_free(pvparser, free);
        ^
2 warnings generated.
gcc -I.. -O3 -std=gnu99 -D __FILTER_DSL_MAIN__ ../containers/sllv.c ../containers/slls.c ../lib/mlrutil.c ../containers/lrec.c ../containers/header_keeper.c filter_dsl_parse.c filter_dsl_lexer.c ..//containers/mlr_dsl_ast.c filter_dsl_wrapper.c -o fdm -lm
filter_dsl_wrapper.c:23:16: warning: implicit declaration of function 'filter_dsl_lemon_parser_parse_token' is invalid in C99 [-Wimplicit-function-declaration]
                parse_code = filter_dsl_lemon_parser_parse_token(pvparser, lex_code, plexed_node, past);
                             ^
filter_dsl_wrapper.c:59:2: warning: implicit declaration of function 'filter_dsl_lemon_parser_free' is invalid in C99 [-Wimplicit-function-declaration]
        filter_dsl_lemon_parser_free(pvparser, free);
        ^
2 warnings generated.
Undefined symbols for architecture x86_64:
  "_filter_dsl_lemon_parser_alloc", referenced from:
      _filter_dsl_parse in filter_dsl_wrapper-edd798.o
  "_filter_dsl_lemon_parser_free", referenced from:
      _filter_dsl_parse in filter_dsl_wrapper-edd798.o
  "_filter_dsl_lemon_parser_parse_token", referenced from:
      _filter_dsl_parse in filter_dsl_wrapper-edd798.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [fdm] Error 1

--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,9 @@
 all: c
 c: .always
-	make -C c install
+	make -C c top
 doc: .always
 	cd doc && poki
+install: .always
+	make -C c install
 .always:
 	@true

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-all: c doc
+all: c
 c: .always
 	make -C c install
 doc: .always

--- a/c/dsls/Makefile
+++ b/c/dsls/Makefile
@@ -13,7 +13,7 @@ put_dsl_wrapper.o: put_dsl_wrapper.c put_dsl_wrapper.h
 put_dsl_parse.o: put_dsl_parse.c put_dsl_parse.h
 	gcc $(CFLAGS) -c -std=gnu99 put_dsl_parse.c
 put_dsl_parse.c put_dsl_parse.h: put_dsl_parse.y lemon
-	lemon put_dsl_parse.y
+	./lemon put_dsl_parse.y
 	mv put_dsl_parse.c put_dsl_parse.c.tmp
 	sed \
 			-e 's/ParseTrace/put_dsl_ParseTrace/g' \
@@ -49,7 +49,7 @@ filter_dsl_wrapper.o: filter_dsl_wrapper.c filter_dsl_wrapper.h
 filter_dsl_parse.o: filter_dsl_parse.c filter_dsl_parse.h
 	gcc $(CFLAGS) -c -std=gnu99 filter_dsl_parse.c
 filter_dsl_parse.c filter_dsl_parse.h: filter_dsl_parse.y lemon
-	lemon filter_dsl_parse.y
+	./lemon filter_dsl_parse.y
 	mv filter_dsl_parse.c filter_dsl_parse.c.tmp
 	sed \
 			-e 's/ParseTrace/filter_dsl_ParseTrace/g' \

